core/impl-base/src/main/java/org/jboss/arquillian/core/impl/loadable/JavaSPIExtensionLoader.java

```diff
@@ -166,7 +167,10 @@ private <T> Set<Class<? extends T>> load(Class<T> serviceClass, ClassLoader load
             Enumeration<URL> enumeration = loader.getResources(serviceFile);
             while (enumeration.hasMoreElements()) {
                 final URL url = enumeration.nextElement();
-                final InputStream is = url.openStream();
+                URLConnection jarConnection = url.openConnection();
+                //Don't cache the file (avoids file leaks on GlassFish).
```

Comment:
```suggestion
                // Avoid caching the file; in combination with Windows OS, can cause file leaks on some EE servers (GF)
                // See WELD-2800 for more details
```

Nitpicking but I'd add a bit more descriptive comment or at least a pointer to the issue which has more details for why it was done.

Reply:
Done. The link to WELD-2800 is actually wrong here, though it is the same issue ;-)

Reply:
Eeh it's my bad, I meant to leave this comment on the Weld PR, not here. I guess I had too many browser tabs open :)
You are right that this one should have the linked Arq. GH issue here.

Reply:
I thought so ;-).
I linked it to this pull request, as there is no issue related to this change. 
Will do the same for the Weld change.